pkg/services/ngalert/notifier/channels/opsgenie.go

```diff
@@ -80,17 +81,44 @@ func NewOpsgenieNotifier(model *NotificationChannelConfig, t *template.Template)
 	}, nil
 }
 
-// Notify sends an alert notification to Opsgenie
 func (on *OpsgenieNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
+	if success, errors := on.NotifyMultiple(ctx, as); !success {
+		count := 0
+		for _, err := range errors {
+			if err != nil {
+				count += 1
+			}
+		}
+		return false, fmt.Errorf("sending Opsgenie notifications on %d alerts", count)
+	}
+
+	return true, nil
+}
+
+func (on *OpsgenieNotifier) NotifyMultiple(ctx context.Context, as []*types.Alert) (bool, []error) {
```

Comment:
I think this function signature can be simplified from `(bool, []error)` to just `[]error` such that if no error occurred we can just return `nil`.